mp2-v1/tests/common/cases/query.rs

```diff
@@ -921,7 +935,182 @@ pub async fn prove_non_existence_row<'a>(
         &planner.pis.bounds,
     );
 
-    let find_node_for_proof = async |query: Option<String>| -> Result<Option<RowTreeKey>> {
+    // this method returns the `NodeContext` of the successor of the node provided as input,
+    // if the successor exists in the row tree and it stores the same value of the input node (i.e., `value`);
+    // returns `None` otherwise, as it means that the input node can be used to prove non-existence
+    async fn get_successor_node_with_same_value(
+        node_ctx: &NodeContext<RowTreeKey>,
+        value: U256,
+        table: &Table,
+        primary: BlockPrimaryIndex,
+    ) -> Option<NodeContext<RowTreeKey>> {
+        let row_tree = &table.row;
+        if node_ctx.right.is_some() {
+            let (right_child_ctx, payload) = row_tree
+                .fetch_with_context_at(node_ctx.right.as_ref().unwrap(), primary as Epoch)
+                .await;
+            // the value of the successor in this case is `payload.min`, since the successor is the
+            // minimum of the subtree rooted in the right child
+            if payload.min() != value {
+                // the value of successor is different from `value`, so we don't return the
+                // successor node
+                return None;
+            }
```

Comment:
Confused because in the comment you say
> if the successor exists in the row tree and it stores the same value of the input node (i.e., `value`); returns `None` 

And yet here the value is different so i would have expected returning something. Same for the other while loop below.

Reply:
We return a node if we find the successor and it has the same value of the input node, that was the intended meaning of the comment; in this case, we found a successor but its value is different from the input node, so we don't return any node. Feel free to modify comments as you wish to clairfy